control/controlclient/auto.go

```diff
@@ -694,6 +694,14 @@ func (c *Auto) StartLogout() {
 func (c *Auto) Logout(ctx context.Context) error {
 	c.logf("client.Logout()")
 
+	// This happens if a user logs out when tailscale client is alreay logged
+	// out either because the user logged out previously or hasn't yet logged
+	// in.
+	if !c.loggedIn {
```

Comment:
Do you need to hold c.mu there?

Reply:
Thank you, yes I guess the auth routine that sets that field can be running concurrently. Updated.